### ISSUE TITLE
 "SQLSTATE[42S21]: Column already exists: 1060 Duplicate column name" on migration command (during install)

### ISSUE THREAD

**Issue by symfonycoda** [aimeos/aimeos-laravel#97]
Hi

I realise this is maybe a general issue to Laravel, but it is affecting me using your package (i have used it in a different project and it is really impressive by the way)!

my issue is that for this project I am using a LavaLiteCMS (https://github.com/LavaLite/cms) nad have now tried to add Aimeos. Everything seems ok until I get to the the migration command, where it picks up that Aimeos is trying to migrate a field called 'city' but there was also a Users field from the CMS package which was called 'city' and so gives me the following error:

  [Illuminate\Database\QueryException]
  SQLSTATE[42S21]: Column already exists: 1060 Duplicate column name 'city' (SQL: alter table `users` add `salutation` varchar(8) not null default '', add `company` varchar(100) not null default '', add
  `vatid` varchar(32) not null default '', add `title` varchar(64) not null default '', add `firstname` varchar(64) not null default '', add `lastname` varchar(64) not null default '', add `address1` var
  char(255) not null default '', add `address2` varchar(255) not null default '', add `address3` varchar(255) not null default '', add `postal` varchar(16) not null default '', add `city` varchar(255) no
  t null default '', add `state` varchar(255) not null default '', add `langid` varchar(5) null, add `countryid` char(2) null, add `telephone` varchar(32) not null default '', add `telefax` varchar(32) n
  ot null default '', add `website` varchar(255) not null default '', add `birthday` date null, add `status` smallint not null default '1', add `vdate` date null, add `editor` varchar(255) not null defau
  lt '')

  [Doctrine\DBAL\Driver\PDOException]
  SQLSTATE[42S21]: Column already exists: 1060 Duplicate column name 'city'

  [PDOException]
  SQLSTATE[42S21]: Column already exists: 1060 Duplicate column name 'city'

Does this mean that I have to change the fieldname and all controller actions and references to it? or is there an easier way? Also if I change these, will they be overriden if your package is upgraded later on?

I have googled but couldn't find an answer.

Many thanks

**Comment by aimeos** [comment]
Which Laravel and Aimeos versions are you using?
Can you also post the database schema of the users table before you run the Aimeos setup?

**Comment by symfonycoda** [comment]
Many thanks.  

Laravel 5.3.*
Aimeos 2017.01 (most recent)




migration to create_users_table : 

<?php

use Illuminate\Database\Migrations\Migration;

class CreateUsersTable extends Migration
{
    /**
     * Run the migrations.
     *
     * @return void
     */
    public function up()
    {
        /*
         * Table: users
         */
        Schema::create('users', function ($table) {
            $table->increments('id')->unsigned();
            $table->integer('reporting_to')->nullable()->default(0);
            $table->string('name', 100)->nullable();
            $table->string('email')->unique();
            $table->string('password', 100);
            $table->string('api_token', 60)->unique();
            $table->string('remember_token', 255)->nullable();
            $table->enum('sex', ['', 'male', 'female'])->nullable();
            $table->date('dob')->nullable();
            $table->string('designation', 50)->nullable();
            $table->string('mobile', 100)->nullable();
            $table->string('phone', 100)->nullable();
            $table->string('address', 255)->nullable();
            $table->string('street', 100)->nullable();
            $table->string('city', 100)->nullable();
            $table->string('district', 100)->nullable();
            $table->string('state', 100)->nullable();
            $table->integer('country')->nullable();
            $table->string('photo', 500)->nullable();
            $table->string('web', 100)->nullable();
            $table->longText('permissions')->nullable();
            $table->enum('status', ['New', 'Active', 'Suspended'])->default('New')->nullable();
            $table->integer('user_id')->nullable();
            $table->string('user_type',50)->nullable();
            $table->string('upload_folder', 100)->nullable();
            $table->softDeletes();
            $table->nullableTimestamps();
        });
    }

    /**
     * Reverse the migrations.
     *
     * @return void
     */
    public function down()
    {
        Schema::drop('users');
    }
}

**Comment by aimeos** [comment]
Can you tell us the steps when this happens?
Is it when `composer update` is executed or `./artisan aimeos:setup`?

**Comment by symfonycoda** [comment]
everything was successful until the `migrate` command and that is when I got the error.

It think because there was a 'city' field in users, and a 'city' field in one of the aimeos migrations as well

**Comment by aimeos** [comment]
Did you update from an earlier version? Because in 2017.01 there are no migration files any more because it's done now via DBAL directly

**Comment by symfonycoda** [comment]
No it was a clean install via composer. 
I had a friend try as well and he came to the same conclusion that the database tables were not compatible. 
he is on Windows and I am on Mac.

I will try again but probably tomorrow evening now.
Thanks for being so prompt. I will update you when I have tried again

**Comment by aimeos** [comment]
We've tried to install Lavalite ourselves (without Aimeos) but with no luck. The `lavalite:install` command always quits with different error messages. I think if Lavalite is set up completely and Aimeos is added afterwards, the `aimeos:setup` command should run without problems and update the `users` table. Only running `aimeos:setup` first may cause problems.

Most columns in the `users` table are named differently than in a standard Aimeos setup. For a good integration, the Aimeos `users` table definition and the customer manager SQL configuration needs to be adapted. Both is easily possible in an extension without the need to touch the Aimeos core.

**Comment by symfonycoda** [comment]
@aimeos I really appreciate your time looking into this. 

Although LaraLiteCMS has been updated to run on Laravel 5.3, it seems that it is not very well maintained. Many of the questions I have for them have been raised on Issues and not answered. They have a PR for a very basic error that I picked up on the install, but the PR has not been merged in 16 days. Their most recent tweet is from Feb 2015! Yes 2015, I just double checked! and I think some of their modules are written in Laravel 4. I will not be continuing with their package, but I have learned a lot about what to check when researching these things. 

Once again thank you for your time. I think your project is great, and I will use it again either with another CMS package or the CMS that I am building for my own use. 

Take care guys and feel free to close this issue

Twitter: @symfonycoda
